azurerm/internal/services/network/firewall_policy_data_source.go

```diff
@@ -0,0 +1,65 @@
+package network
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmFirewallPolicy() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmFirewallPolicyRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
```

Comment:
I'm not sure what tge point of this datasource is? none of the other computed properties are on it, so could we remove it or add more to make it useful?

Reply:
I was following the practice that we implement an "empty" data source at the beginning (not really useless as user can get the id from the input argument and refer to the ID in other resources), then add useful attributes when necessary.
So do you think we shall always fill in the available attributes into data source at the beginning? If so, then I'll change my practice and complement the missing attributes in this data source.

Reply:
At the very least we can throw the computed values in here as those make somewhat sense, and if theID is important/something they could need the ds also makes sense. but i'm not sure if that is the case here?

Reply:
Never mind, I have supported it anyway 🙃